HEN_HOUSE/egs++/geometry/egs_cd_geometry/egs_cd_geometry.h

```diff
@@ -506,9 +506,25 @@ class EGS_CDGEOMETRY_EXPORT EGS_CDGeometry : public EGS_BaseGeometry {
                     }
                     return ixold;
                 }
-                else if (ixnew < 0 && tb <= epsilon) {                         // (a) is true
+				// Skip for concave geometries: particles can reenter the CD geometry after leaving.
+				// Check 2. below will catch these cases.
+                else if ((ixnew < 0 && tb <= epsilon) && (bg->isConvex() && g[ibase]->isConvex())) {                         // (a) is true
```

Comment:
In this logical if one must check whether `g[ibase] `exists or `icd = 0` to avoid a **segfault** error in cases where only one geometry is inscribed in one of the regions of the base geometry.